### DISCUSSION TITLE
Enhancing the reception of RX470C 433 MHz receiver modules

### DISCUSSION THREAD

**Discussion by de-dh** [micropython/micropython#16232] [Show and tell]
I discovered Peter Hinch's micropython_remote recebtly and experimented with rf modules. While sending signals is easy with the Esp32 rmt-class, receiving data proved to be more difficult. This is mainly attributed to high noise levels of low-cost receivers. I found a simple way to increase the quality of reception specifically for RX470C 433 MHz receivers which I want to share with you. Maybe it is helpful to someone.

I am using RX470C receivers which are based on the SYN470R IC or a similar copy.
I discovered a simple way to to reduce noise and increase quality of reception. By adding a several Megaohm resistor to specific pins of the IC analog noise reduction can be achieved and timing values of the signals become more precise. I've attached an image showing the neccessary modifications. In my case a 1 MOhm resistor was sufficient and I also attached a supply bypass capacitor.

Consequently, noise is almost completely eradicated and DO signal ist only present
when a legitimate transmission is detected. 
I wrote a program which can receive and decode the signals of a ThermoPro TP65S outdoor sensor. I've analyzed the RF transmissions of the sensor unsig a HackRF. The relevant low pulses have durations of approximately 2000us and 4000us. Using the RX470C module without modifications, I picked up average pulse lengths of approx. 1700 and 3700us. After I attached the resistor, timing values became more accurate with approx. values of 1900 us and 3900 us.

I wrote a more detailed copy of my findings [here](https://github.com/de-dh/MPY-RF) ncluding a simple receiver program.
![RX470C_modified](https://github.com/user-attachments/assets/6d898203-4215-449b-a66f-6ed799b86925)

**Comment by bidrohini** [comment]
Thanks for sharing. Can we use generic RF modules instead? Like the red one shown here?

https://www.theengineeringprojects.com/2023/04/interface-remote-control-rf-module-433mhz-with-pi-4.html

**Comment by de-dh** [comment]
It depends on the RF receiver's IC.
I only know for sure that it works with the SYN470R IC as described in the [datasheet](https://img.ozdisan.com/ETicaret_Dosya/453934_1928180.pdf) in Sec. 14.5.
The problem is, that the receiver ICs are often unlabelled so it's hard to tell which IC your receiver uses.
If you want to make sure, you can buy a RX470C module. They are often sold in pair with WL102-341 transmitter modules for 2 - 3€.
I bought a set of five pairs of RX470C + WL102-341 + antennas on amazon for ~13€. They are a little bit cheaper on ALiExpress.


However, if you want to build the circuits described on the link you provided, I don't think you need those modifications.
Those circuits use HT12E ICs for encoding / decoding so that you don't need to deal with the rf data in your software.
You will simply connect the output pins of the HT12E IC in the receiver circuit to your RPI's GPIO pins and the appropriate Data Pin will be high when a signal is received.  
E.g. Transmitter channel A1 is high --> Receiver D1 will be high

**Reply by bidrohini** [reply]
Thanks a lot. Your post is very helpful.

**Comment by mendenm** [comment]
I notice on the dat sheet that CTH is supposed to have a capacitor to ground, which will float and automatically set a threshold.  The pin is definitely not supposed to be left open.  Did you try a capacitor, per the  sample circuit in the spec sheet?

**Comment by de-dh** [comment]
No, I didn't try that.
I think it already has the capacitor on the RX470C module.
I found a circuit diagram of the module [here](https://qiachip.com/en-de/products/qiachip-433mhz-superheterodyne-rf-receiver-and-transmitter-module?srsltid=AfmBOop_2uBDeD1v2y2sE0ECEibIyCgXrLNbrzRt97OA_lCqBPjymwUw).
